Lib/typing.py

```diff
@@ -178,7 +180,11 @@ def _type_check(arg, msg, is_argument=True, module=None, *, allow_special_forms=
 
     We append the repr() of the actual value (truncated to 100 chars).
     """
-    invalid_generic_forms = (Generic, Protocol)
+    try:
+        invalid_generic_forms = (Generic, Protocol)
+    except NameError:
+        # Run during interpreter startup
```

Comment:
nit: if we keep the current catch-NameError approach to handling this cycle (see below for an alternative), I think this comment could be more accurate and informative. In my local testing, I don't think it is necessarily called during interpreter startup, only whenever the `typing` module is imported.

```suggestion
        # Called indirectly by `class _Dummy[...]` below, before Generic is bound
```

Reply:
This is no longer an issue with the `_typing` approach.